src/app/revenue-share/revenue-share.component.ts

```diff
@@ -1,26 +1,100 @@
+        this.fetchDataFromRoute();
+        this.formatConfirmPopupInputField();
+    }
 
-  fetchDataFromRoute() {
-    this.projectID = this.route.snapshot.params.projectID;
-    this.amount = this.route.snapshot.params.amount;
-  }
+    fetchDataFromRoute() {
+        this.projectID = this.route.snapshot.params.projectID;
+        this.amountInvested = this.route.snapshot.params.amount;
+        this.getProject(this.projectID);
+    }
 
+    getProject(projectUUID: string) {
+        SpinnerUtil.showSpinner();
+        this.projectService.getProject(projectUUID)
+            .subscribe(res => {
+                SpinnerUtil.hideSpinner();
+                this.projectName = res.name;
+            }, error => {
+                hideSpinnerAndDisplayError(error);
+            });
+    }
+
+    confirmButtonClicked() {
+        const revenueAmountInvested = parseInt(stripCurrencyData(String(this.amountInvested)), 10);
+        const revenueAmountInvestedConfirm = parseInt(stripCurrencyData(String(this.amountInvestedConfirm)), 10);
+
+        if (revenueAmountInvested !== revenueAmountInvestedConfirm) {
+            swal('', 'The revenue share amounts don\'t match. Please check the proper amount and try again!',
+                'error').then(() => {
+                (<any>$('#modal-confirm-revenue')).modal('hide');
+                location.reload();
+            });
+            return;
+        }
+        this.generateTransactionForRevenuePayout(revenueAmountInvestedConfirm);
+    }
+
+    generateTransactionForRevenuePayout(amountInvested: number) {
+        SpinnerUtil.showSpinner();
+        this.managePaymentService.generateTransactionForRevenuePayout(this.projectID, amountInvested)
+            .subscribe(async (res: any) => {
+                this.closeConfirmPopupDialog();
+                const arkaneConnect = new ArkaneConnect('AMPnet', {environment: 'staging'});
+                const acc = await arkaneConnect.flows.getAccount(SecretType.AETERNITY);
+                const sigRes = await arkaneConnect.createSigner(WindowMode.POPUP).sign({
+                    walletId: acc.wallets[0].id,
+                    data: res.tx,
+                    type: SignatureRequestType.AETERNITY_RAW
+                });
+                this.broadcastService.broadcastSignedTx(sigRes.result.signedTransaction, res.tx_id)
+                    .subscribe(_ => {
+                        swal('', 'Successful revenue payout!', 'success');
+                        SpinnerUtil.hideSpinner();
+                        this.closeConfirmPopupDialog();
+                    }, err => {
+                        hideSpinnerAndDisplayError(err);
+                    });
+            }, err => {
+                hideSpinnerAndDisplayError(err);
+            });
+    }
+
+    closeConfirmPopupDialog() {
+        this.modalPopupDialog.nativeElement.click();
+    }
+
+    formatConfirmPopupInputField() {
+        autonumericCurrency('#revenue-confirm-amount');
+    }
```

Comment:
Can this be a pipe? If this is the way the entire app works, it's OK for now, but it shouldn't be made this way. We'll talk about it later.

Reply:
I agree, but I would leave it that for now, because that’s how we use it in many of our components